### PR TITLE
DXCDT-122: Adding deletion E2E test cases

### PR DESCRIPTION
**Pull request by willvedd**
## ✏️ Changes

Building off of three previous E2E testing PRs (#610, #611 , #615) to bring behavior tests for deletions. Specifically, testing that deletions of resources do work and that the tool respects the `AUTH0_ALLOW_DELETE` boolean environment variable. The bulk of the work here was introducing sets of configuration to accurately test deletions and also the HTTP recordings. There were some other miscellaneous changes that were made to further improve the E2E testing as well as the request recording process.

This represents the final task of this current testing improvement initiative. 

## 🎯 Testing

This PR adds E2E testing for resource deletion.

### CODE REVIEW

File: .prettierignore [hunk 1]
```diff
@@ -7,4 +7,5 @@ lib
 node_modules
 examples
 test/**/*.json
-test/**/*.yaml
\ No newline at end of file
+test/**/*.yaml
+test/e2e/testdata
```
Comment (willvedd) [inline .prettierignore:11]: Need to ignore code formatting for the test configuration data otherwise the HTTP recordings won't precisely match.

File: .eslintignore [hunk 2]
```diff
@@ -1,4 +1,5 @@
 examples/
+test/e2e/testdata/**/**
```
Comment (willvedd) [inline .eslintignore:2]: Need to ignore linting for the test configuration data otherwise the HTTP recordings won't precisely match.

File: test/e2e/e2e-utils.ts [hunk 3]
```diff
@@ -5,11 +5,10 @@ import { back as nockBack, BackMode, Definition } from 'nock';
 
 export function decodeBuffer(recordingResponse: Definition['response']) {
   try {
-    // Decode the hex buffer that nock made
-    const response = Array.isArray(recordingResponse)
-      ? recordingResponse.join('')
-      : recordingResponse?.toString() || '';
-    const decoded = Buffer.from(response, 'hex');
+    const isArray = Array.isArray(recordingResponse);
+    if (!isArray) return recordingResponse; // Some Auth0 Management API endpoints aren't gzipped, we can tell this if they are an array or not
```
Comment (willvedd) [inline test/e2e/e2e-utils.ts:9]: From my tests, there appear to be some Management API endpoints that do not gzip their payloads, so when attempting to decode for the purposes of recording, it would fail. I found that this check if the payload is an array is a simplistic way of determining wether to decode the payload. If it is not an array of hex bytes, then it its likely already a parsed payload. This check is probably imperfect and susceptible to edge cases, but it works for now.

File: test/e2e/e2e-utils.ts [hunk 4]
```diff
@@ -50,6 +49,7 @@ export function sanitizeRecording(recording: Definition): Definition {
     'client_secret',
     'cert',
     'pkcs7',
+    'key',
```
Comment (willvedd) [inline test/e2e/e2e-utils.ts:52]: Need to ignore the private key property for signing keys.

File: .prettierignore (hunk 1 above)
Comment (sergiught) [inline .prettierignore:11]:
```suggestion
test/e2e/testdata

```

File: test/e2e/testdata/lots-of-configuration/pages/login.html [hunk 5]
```diff
@@ -0,0 +1 @@
+<html>TEST123</html>
```
Comment (sergiught) [inline test/e2e/testdata/lots-of-configuration/pages/login.html:1]:
```suggestion
<html>TEST123</html>

```

File: test/e2e/testdata/lots-of-configuration/rules/my-rule.js [hunk 6]
```diff
@@ -0,0 +1,3 @@
+function (user, context, callback) {
+  callback(null, user, context);
+}
```
Comment (sergiught) [inline test/e2e/testdata/lots-of-configuration/rules/my-rule.js:3]:
```suggestion
}

```

**Review by sergiught — APPROVED** [review]
Nice work 👍🏻
